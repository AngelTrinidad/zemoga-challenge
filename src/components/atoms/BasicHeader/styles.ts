import {StyleSheet} from 'react-native';

import {DEFAULT_SPACES} from '~constants/styles';
import {scaleSpace} from '~helpers';

const styles = StyleSheet.create({
  header: {
    paddingHorizontal: DEFAULT_SPACES.paddingScreen,
    paddingVertical: scaleSpace(24),
  },
});

export default styles;
