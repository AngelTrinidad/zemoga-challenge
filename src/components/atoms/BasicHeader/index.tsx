import React, {FC, PropsWithChildren} from 'react';
import {View, ViewStyle} from 'react-native';

import styles from './styles';

import {useTheme} from '~contexts';

export interface BasicHeaderProps extends PropsWithChildren {
  style?: ViewStyle;
}

const BasicHeader: FC<BasicHeaderProps> = ({children, style}) => {
  const {colors} = useTheme();

  const headerStyles: ViewStyle = {
    backgroundColor: colors.secondary,
  };

  return <View style={[styles.header, headerStyles, style]}>{children}</View>;
};

export default BasicHeader;
