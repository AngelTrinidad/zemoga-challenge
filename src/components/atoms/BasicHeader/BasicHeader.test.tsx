import React from 'react';
import {Text} from 'react-native';

import BasicHeader from '.';

import {render} from '~helpers/testing';

describe('<BasicHeader />', () => {
  let component: any;

  beforeEach(() => {
    component = render(
      <BasicHeader>
        <Text>Hello</Text>
      </BasicHeader>,
    );
  });

  it('should render correctly', () => {
    expect(component).toBeDefined();
  });

  it('should has same snapshot', () => {
    expect(component.toJSON()).toMatchSnapshot();
  });
});
