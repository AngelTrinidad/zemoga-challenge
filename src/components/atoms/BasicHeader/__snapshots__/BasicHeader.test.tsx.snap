// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<BasicHeader /> should has same snapshot 1`] = `
<View
  style={
    [
      {
        "paddingHorizontal": 22.857142857142858,
        "paddingVertical": 27.42857142857143,
      },
      {
        "backgroundColor": "#CFB997",
      },
      undefined,
    ]
  }
>
  <Text>
    Hello
  </Text>
</View>
`;
